### PR TITLE
fix(install): Fix Junction creation during installation inside containers

### LINKED ISSUE #4792: [Bug] scoop in docker shims seem broken
#### Current Behavior
install scoop in a docker container
install a scoop package (such as git)
use scoop package in a subsequent RUN command results in:
```
Could not create process with command '"C:\Users\ContainerAdministrator\scoop\apps\git\current\cmd\git.exe"  clone -q https://github.com/samuelcolvin/pydantic.git C:\Users\ContainerAdministrator\AppData\Local\Temp\pip-install-xmwtmaol\pydantic_4641e039e3774955b81c25db0d639f23'.
```

#### Expected Behavior
Can use installed programs

#### Possible Solution
https://github.com/ScoopInstaller/Scoop/tree/2021-12-26 does work

### System details

**Windows version:** 10

**OS architecture:** 64bit

**PowerShell version:** [output of `"$($PSVersionTable.PSVersion)"`]

**Additional software:** [(optional) e.g. ConEmu, Git]

#### Scoop Configuration

```json
//# Your configuration here
```

I will try to add more config details asap, but maybe someone already has an idea?

### PR DESCRIPTION
**Pull request by kingster**
#### Description
Fixes installation of scoop in Dockerfile due to https://github.com/docker/for-win/issues/12240

#### Motivation and Context

Fixes #4792

#### How Has This Been Tested?

Tested in windows docker & windows vm

#### Checklist:

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

**Comment by kingster** [comment]
@rashil2000 Can you have a look at this? 

I couldn't figure out how to test this, please point me to any doc which I can follow to install from the repo directly.

File: lib/install.ps1 [hunk 1]
```diff
@@ -1270,3 +1268,26 @@ function test_running_process($app, $global) {
...
+function Test-IsInsideContainer {
+    $foundService = Get-Service -Name cexecsvc -ErrorAction SilentlyContinue
+    if ( $foundService -eq $null ) {
+        $false
+    }
+    else {
+        $true
+    }
+}
+
```
Comment (rashil2000) [inline lib/install.ps1:1282]:
```suggestion
```

**Comment by rashil2000** [comment]
Also, please keep the `attrib` commands in their original locations (instead of inside new function)

**Comment by rashil2000** [comment]
Hmm, looks good. One minor nit: We should keep the `Out-Null` part in the calling location itself, as the output might be required if we were to reuse the New-DirectoryJunction function somewhere else.

Also, please update the changelog

**Comment by kingster** [comment]
Done!

**Review by rashil2000 — APPROVED** [review]

**Review by niheaven — APPROVED** [review]

**Pull request merged**